system/internal/service/user.go

```diff
@@ -163,14 +171,39 @@ func (svc *user) Update(mod *types.User) (u *types.User, err error) {
 		u.Handle = mod.Handle
 		u.Kind = mod.Kind
 
+		// Store avatar for user
+		if u, err = svc.bindAvatar(u, avatar, avatarURL); err != nil {
+			return err
+		}
+
 		if u, err = svc.user.Update(u); err != nil {
+			svc.unbindAvatar(u)
 			return err
 		}
 
 		return nil
 	})
 }
 
+func (svc *user) bindAvatar(user *types.User, avatar *multipart.FileHeader, avatarURL string) (*types.User, error) {
```

Comment:
this is not ok, we should not be dependant on http inside service layer. 
file should be resolved inside controller and service should get io.Reader (or a similar one from that family)